test/DynamoCoreWpfTests/WorkspaceSaving.cs

```diff
@@ -941,5 +941,44 @@ public void CusotmNodeSaveAsUpdateItsName()
             }
         }
         #endregion
+
+        #region Recent Files
+        [Test]
+        public void SavingFilesUpdateRecentFileList()
+        {
+            // Open a file
+            var examplePath = Path.Combine(TestDirectory, @"core\math", "Add.dyn");
+            ViewModel.OpenCommand.Execute(examplePath);
+
+            List<string> paths = new List<string>();
+            
+            ViewModel.Model.PreferenceSettings.MaxNumRecentFiles = 3;
+            // Save the file as different files for (maxNum + 1) times
+            int maxNum = ViewModel.Model.PreferenceSettings.MaxNumRecentFiles;
+            for (int i = 0; i < maxNum + 1; i++)
+            {
+                var newPath = GetNewFileNameOnTempPath("dyn");
+                var res = ViewModel.Model.CurrentWorkspace.SaveAs(newPath, ViewModel.Model.EngineController.LiveRunnerRuntimeCore);
+                Assert.IsTrue(res);
+                paths.Add(newPath);
+            }
+
+            // Ensure the number of recent files reaches the maximum number
+            Assert.AreEqual(maxNum, ViewModel.Model.PreferenceSettings.RecentFiles.Count);
```

Comment:
When you are checking for `PreferenceSettings.RecentFiles` I believe you are expecting it to have updated when the observable collection property `RecentFiles` in `DynamoViewModel` changes due to this:
```
this.RecentFiles.CollectionChanged += (sender, args) =>
            {
                model.PreferenceSettings.RecentFiles = this.RecentFiles.ToList();
            };
```
Is that right?